src/bundle/Resources/public/js/scripts/admin.context.menu.js

```diff
@@ -21,40 +25,109 @@
+                const relatedSubitemBtnId = subitemBtn.id;
 
-    popupMenu.generateItems(popupItemsToGenerate, (itemElement, item) => {
-        const itemContentElement = itemElement.querySelector('.ibexa-popup-menu__item-content');
+                return {
+                    label: subitemLabel,
+                    onClick: () => clickRelatedBtn(relatedSubitemBtnId),
+                    processAfterCreated: (itemElement) => {
+                        const itemBtn = itemElement.querySelector('.ibexa-multilevel-popup-menu__item-content');
 
-        itemElement.dataset.relatedButtonId = item.relatedButtonId;
+                        itemBtn.disabled = subitemBtn.disabled;
+                        addRelatedBtnIdToMenuItem(itemElement, relatedSubitemBtnId);
+                    },
+                };
+            });
 
-        if (item.disabled) {
-            itemContentElement.classList.add('ibexa-popup-menu__item-content--disabled');
+            return {
+                label: alternativeToggleLabel ?? mainBtnLabel,
+                onClick: () => clickRelatedBtn(relatedMainBtnId),
+                processAfterCreated: (itemElement) => {
+                    const itemBtn = itemElement.querySelector('.ibexa-multilevel-popup-menu__item-content');
+
+                    itemBtn.disabled = mainBtn.disabled;
+                    addRelatedBtnIdToMenuItem(itemElement, relatedMainBtnId);
+                },
+                branch: {
+                    groups: [
+                        {
+                            id: 'main',
+                            items: [
+                                {
+                                    label: mainBtnAlternativeLabel ?? mainBtnLabel,
+                                    sublabel: mainBtnAlternativeSublabel,
+                                },
+                            ],
+                        },
+                        {
+                            id: 'subitems',
+                            items: subitems,
+                        },
+                    ],
+                },
+            };
         }
+
+        const relatedBtnId = menuButton.id;
+        const label = menuButton.querySelector('.ibexa-btn__label').textContent;
+
+        return {
+            label,
+            groupId: 'default',
+            onClick: () => clickRelatedBtn(relatedBtnId),
+            processAfterCreated: (itemElement) => addRelatedBtnIdToMenuItem(itemElement, relatedBtnId),
+        };
     });
+    const menuTree = {
+        triggerElement: showPopupButton,
+        placement: 'bottom-end',
+        fallbackPlacements: ['bottom-start', 'top-end', 'top-start'],
+        groups: [
+            {
+                id: 'default',
+                items: topBranchItems,
+            },
+        ],
+    };
+
+    multilevelPopupMenu.init();
+    const topBranch = multilevelPopupMenu.generateMenu(menuTree);
```

Comment:
```suggestion
    
    const topBranch = multilevelPopupMenu.generateMenu(menuTree);
```

Reply:
I will change it, but I actually like the current notation more.